pennylane/math/matrix_manipulation.py

```diff
@@ -224,35 +161,122 @@ def _sparse_expand_matrix(base_matrix, wires, wire_order, format="csr"):
     Returns:
         tensor_like: expanded matrix
     """
-    n_wires = len(wires)
-    n_total_wires = len(wire_order)
+    base_matrix.eliminate_zeros()
+    wires = wires.tolist() if isinstance(wires, qml.wires.Wires) else list(copy.copy(wires))
+
+    # get a subset of `wire_order` values that contain all wire labels inside `wires` argument
+    # e.g. wire_order = [0, 1, 2, 3, 4]; wires = [3, 0, 2]
+    # --> subset_wire_order = [0, 1, 2, 3]; expanded_wires = [3, 0, 2, 1]
+    wire_indices = [wire_order.index(wire) for wire in wires]
+    subset_wire_order = wire_order[min(wire_indices) : max(wire_indices) + 1]
+    wire_difference = list(set(subset_wire_order) - set(wires))
+    expanded_wires = wires + wire_difference
+
+    # expand matrix if needed
+    if wire_difference:
+        base_matrix = kron(base_matrix, eye(2 ** len(wire_difference), format="coo"), format="csr")
+        base_matrix.eliminate_zeros()
+
+    # permute the matrix to match the `subset_wire_order` values
+    U = _permutation_sparse_matrix(expanded_wires, subset_wire_order)
+    if U is not None:
+        base_matrix = U.T @ base_matrix @ U
+        base_matrix.eliminate_zeros()
+
```

Comment:
Since we did all that nice work sorting the wires above:

```
mat = base_matrix
num_pre_identities = min(wire_indices)
if num_pre_identities > 0:
    pre_identity = eye(2**num_pre_identities, format='csr')
    mat = kron(pre_identity, mat)

num_post_identities = len(wire_order) - max(wire_indices) - 1
if num_post_identities > 0:
    post_identity = eye(2**num_post_identities, format='csr')
    mat = kron(mat, post_identity)
```